---

# Actividad 2 Ciencia de Datos

### 1. Selección de Tema y Fuente de Información

**Tema Elegido**: Inversión Sostenible y Energías Renovables

La inversión sostenible y las energías renovables son cruciales para el desarrollo sostenible, alineándose directamente con varios Objetivos de Desarrollo Sostenible (ODS) de la Agenda 2030, como el ODS 7 (Energía asequible y no contaminante), ODS 13 (Acción por el clima), y ODS 11 (Ciudades y comunidades sostenibles). La transición hacia energías limpias y la inversión en tecnologías sostenibles son fundamentales para reducir las emisiones de gases de efecto invernadero y construir una economía resiliente al clima.

**Fuente de Información**: Yahoo Finance API (`yfinance` en Python)

`yfinance` es una biblioteca de Python que permite acceder fácilmente a datos financieros históricos, cotizaciones en tiempo real, datos fundamentales de empresas, y mucho más, proporcionados por Yahoo Finance.

### 2. Descripción de Datos y Metadatos

La API `yfinance` ofrece una amplia variedad de datos sobre acciones, incluidas cotizaciones históricas, información sobre dividendos y splits, resúmenes financieros, entre otros. Los metadatos asociados con estos datos incluyen el ticker de la acción, rango de fechas de las cotizaciones, volumen negociado, precios de apertura/cierre máximo/mínimo, entre otros. Estos metadatos son esenciales para el análisis financiero y pueden ser utilizados para evaluar el desempeño de inversiones en energías renovables y sostenibilidad.

Los metadatos proporcionados por `yfinance`, como el volumen negociado y los precios de apertura/cierre, son fundamentales para evaluar la estabilidad y el crecimiento a largo plazo de las inversiones en energías renovables. Esta información, al ser correlacionada con indicadores de sostenibilidad (ESG), permite identificar tendencias de inversión sostenible y evaluar su alineación con los ODS, particularmente en lo que respecta a la innovación y el impacto económico sostenible (ODS 9).

### 3. Funcionamiento de la API `yfinance`

`yfinance` funciona mediante la creación de un objeto que representa una acción o un conjunto de acciones, utilizando los tickers del mercado. Desde este objeto, se pueden solicitar diferentes tipos de datos: históricos, fundamentales, etc. La biblioteca maneja internamente las solicitudes HTTP a los servidores de Yahoo Finance y parsea los datos recibidos en formatos fácilmente manejables en Python, como DataFrames de pandas.


### 4. Desarrollo del Programa en Python

El objetivo será obtener datos de fondos líderes en inversión sostenible (por ejemplo, S&P 500 ESG Index - ESG) y realizar un análisis sencillo, como la visualización de la tendencia de su precio de cierre en los últimos años y calcular su retorno medio.



### 5. Análisis y Discusión de Resultados

Tras implementar y ejecutar el programa, se discutirán los resultados obtenidos, como la tendencia en el precio de la acción de una empresa de energías renovables, lo que puede reflejar la creciente adopción e interés en la sostenibilidad. También, se discutirá el cálculo del retorno medio como indicador del desempeño financiero de la inversión sostenible.

Las herramientas de visualización como Seaborn y Matplotlib serán utilizadas para representar gráficamente las tendencias de inversión en energías renovables y su impacto en los ODS. Por ejemplo, gráficos de líneas mostrando el crecimiento a largo plazo del ETF iShares Global Clean Energy frente a indicadores de desarrollo sostenible, proporcionarán insights visuales sobre cómo las inversiones sostenibles están contribuyendo a los objetivos globales de sostenibilidad.


---

# Objetivo del análisis

Este análisis busca comparar la inversión en el S&P 500, conocido como el índice más invertido del mundo, con alternativas más responsables, como los fondos e índices ESG (Ambiental, Social y Gobernanza). Mientras que el S&P 500 ofrece una visión general de las empresas de gran capitalización en EE. UU., los índices ESG y fondos seleccionados, como el S&P 500 ESG, iShares ESG Aware MSCI USA ETF (ESGU), y iShares ESG Aware MSCI USA ETF (SUSA), representan opciones de inversión que consideran factores de sostenibilidad junto al rendimiento financiero, reflejando el creciente interés por combinar responsabilidad social y ambiental con inversión.

## Índices y ETFs seleccionados para el análisis:

### S&P 500 ESG Index
El S&P 500 ESG Index es un índice de mercado amplio y ponderado por capitalización de mercado que busca medir el rendimiento de los valores que cumplen con criterios de sostenibilidad. Este índice mantiene pesos similares en los grupos de la industria en comparación con el S&P 500, proporcionando una manera de seguir comprometido con las inversiones sostenibles mientras se mantiene una exposición similar a la del mercado general.

### iShares ESG Aware MSCI USA ETF (ESGU)
El iShares ESG Aware MSCI USA ETF busca replicar el rendimiento del índice MSCI USA Extended ESG Focus Index. Este fondo invierte en empresas estadounidenses de gran y mediana capitalización que tienen altos puntajes ESG en relación con sus pares sectoriales. Aunque no encontré una descripción detallada específica para ESGU en las fuentes citadas, se puede inferir que, como parte de la familia de productos iShares y siguiendo criterios ESG, busca proporcionar una inversión responsable en términos ambientales, sociales y de gobernanza.

### iShares ESG Aware MSCI USA ETF (SUSA)
El iShares MSCI USA ESG Select ETF selecciona compañías con altas calificaciones ESG en comparación con sus pares sectoriales y excluye actividades empresariales controvertidas como el alcohol, tabaco, juegos de azar, armas civiles, armas nucleares y carbón térmico. El ETF incorpora cuestiones clave como el cambio climático, el capital natural, la contaminación y los residuos, el capital humano, la responsabilidad del producto, la gobernanza corporativa y la oposición de los interesados en sus decisiones de inversión. Este fondo es reconstituido y reequilibrado semianualmente

Estos instrumentos ofrecen opciones para los inversores interesados en promover prácticas sostenibles y responsables a través de sus carteras de inversión, permitiendo al mismo tiempo una diversificación similar a los índices de mercado amplios. Cada uno de estos productos tiene un enfoque particular en integrar consideraciones ambientales, sociales y de gobernanza en el proceso de selección de inversiones, reflejando el creciente interés en las inversiones sostenibles y el deseo de impactar positivamente en el mundo a través de la inversión.

In [ ]:
import yfinance as yf
import pandas as pd
import plotly.graph_objects as go

# Define los tickers de los fondos ESG y los índices
tickers = ['SPY', 'ESG', 'SUSA', 'ESGU']

# Obtiene los datos históricos de los últimos 5 años
data = yf.download(tickers, start='2019-01-01', end='2024-01-01')['Adj Close']

# Calcula la rentabilidad acumulada
rentabilidad_acumulada = (data / data.iloc[0] * 100)

# Crea la gráfica
fig = go.Figure()

# Añade las líneas a la gráfica
for ticker in tickers:
    fig.add_trace(go.Scatter(x=rentabilidad_acumulada.index, y=rentabilidad_acumulada[ticker], mode='lines', name=ticker))

# Actualiza el layout de la gráfica para añadir título y labels
fig.update_layout(title='Rentabilidad Acumulada de los Fondos ESG vs. S&P 500 y S&P 500 ESG (Últimos 5 Años)',
                  xaxis_title='Fecha',
                  yaxis_title='Rentabilidad Acumulada (%)',
                  hovermode='x unified')

# Hace que la leyenda alterne las series de datos al hacer clic
fig.update_layout(legend=dict(itemsizing='constant', traceorder='normal', title='Haga clic para mostrar/ocultar:'))

# Muestra la gráfica
fig.show()


[*********************100%%**********************]  4 of 4 completed


In [ ]:
# Calcula la rentabilidad desde el inicio hasta el final del periodo para cada ticker
rentabilidad = ((data.iloc[-1] - data.iloc[0]) / data.iloc[0]) * 100

# Convierte la rentabilidad en un DataFrame para una visualización clara
tabla_rentabilidad = pd.DataFrame(rentabilidad, columns=['Rentabilidad (%)'])


print(tabla_rentabilidad)



        Rentabilidad (%)
Ticker                  
ESG           108.169650
ESGU          106.164885
SPY           106.156873
SUSA          109.832214



3. **Análisis y Discusión**:


Los resultados muestran que las inversiones en fondos e índices ESG (ESG, ESGU, y SUSA) han tenido un rendimiento ligeramente superior o comparable al del S&P 500 (SPY) en los últimos 5 años. SUSA lidera con una rentabilidad del 109.83%, seguido por ESG con 108.17%. ESGU y SPY muestran un rendimiento similar, con 106.16% y 106.16% respectivamente. Esto sugiere que las inversiones responsables no solo son competitivas en términos de rendimiento, sino que también pueden ofrecer retornos ligeramente superiores, reforzando el caso para considerar factores ESG en decisiones de inversión.